### DISCUSSION TITLE
Timestamp precision in nanoseconds

### DISCUSSION THREAD

**Discussion by apeksha-r-naik** [finos/TimeBase-CE#266] [Q&A]
Hi @Romkisel @alex-karpovich, 

We are using timebase client 6.0.5

And we are inserting the timestamp in Time base with Nanoseconds level precision, but when we query the time base we are getting the timestamp in Milliseconds.

For an Example: 

Python code while loads the data in Time base ( timestamp is in nanos ) : 1705388591200091000

When we query Time base we are getting (in milliseconds) : 2024-01-16T07:03:11.200Z

Expected format is (in nanoseconds)  : 2024-01-16T07:03:11.200091000Z

Please suggest us further. Thanks!

**Comment by alex-karpovich** [comment]
Greetings, @apeksha-r-naik 
Are you using custom field or system field 'timestamp'?

**Reply by ankitsoin** [reply]
System field timestamp.

**Reply by alex-karpovich** [reply]
Greetings, @ankitsoin 

timestamp field in QQL mapped to getTimeStamp(), we will add timestampns field in next release which will return nanoseconds
